theano/scan_module/scan_opt.py

```diff
@@ -351,14 +378,14 @@ def process_node(self, fgraph, node):
                 remove=[node],
                 reason='scanOp_pushout_nonseqs_ops')
             return True
-        elif to_keep == []:
+        elif not to_keep:
```

Comment:
This version reduces readability compared to the previous and I would be surprised if it made a significant impact on performance so it should probably be left as it was. The code for Scan seriously lacks readability so any small thing can make a big difference in that regard.